services/app/src/components/chessBoard/Square.jsx

```diff
@@ -18,6 +18,7 @@ const Square = ({
 }) => {
   const { board, currentPlayerId } = useChessBoardContext();
   const [showPopover, setShowPopover] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
```

Comment:
`isHovered` and `showPopover` is essentially the same the same thing. Maybe we can drop `showPopover` since it shows only when user hovers on square.

Reply:
The problem with this is that `showPopover` controls only the placement of the piece (and the tiles where the piece is placed), but it doesn't account for the hover of tiles. Therefore, the is no way to check for hover universally:

1. If I do showPopover, it only displays values when we hover on the occupied tile
2. If I do !showPopover, it displays values on all tiles but the occupied tile without even hovering over them

Therefore, we need a way to universally change the display of values along with the mouse actions (in this case - hovering). `isHovered` solves this problem by letting us check whether the tile is hovered and whether a piece is placed there or not:

```
onMouseEnter={() => {updatePopover(true); setIsHovered(true)}}
onMouseLeave={() => {updatePopover(false); setIsHovered(false)}}
```

Reply:
I meant deleting ```showPopover``` and keeping ```isHovered```. Your logic for checking if those attributes(numbers with letters) are needed to be shown is good. 
As for showing a popover, you can pass ```isHovered``` to ```SquarePopUp``` component, where you can check if displaying of popover is needed or not.